### PR TITLE
Adds ability to cull (remove) per-publisher items on a periodic, poli…

### LINKED ISSUE #277: Remove hub content on a periodic basis

### PR DESCRIPTION
**Pull request by richardrodgers**
…cy-driven basis.

Closes #277

### CODE REVIEW

File: app/controllers/Application.scala [hunk 1]
```diff
@@ -330,6 +331,110 @@ object Application extends Controller with Security {
...
+      Ok("kicked off all culls")
+    } else {
+      Unauthorized(views.html.static.trouble("You are not authorized"))
+    }
+  }
+
+  def expungeKnownItem(cid: Int, oid: String, policy: String) =
+    isAuthenticated { identity => implicit request =>
+    Collection.findById(cid).map ( coll => {
+      if (coll.publisher.userId == identity.id) {
```
Comment (JPrevost) [inline app/controllers/Application.scala]: This is a general comment, and not specific to this PR as we can make this change later if we agree with my reasoning: after our last planning meeting I came away with the idea that Harvests (and Culls now) should not be tied to a Publisher's User. Instead, Hub Analysts or SysAdmins should be required for these functions (which actually simplifies these methods).

File: app/models/Item.scala [hunk 2]
```diff
@@ -29,6 +29,20 @@ case class Item(id: Int,            // DB key
...
                 transfers: Int)  {  // Number of transfers
 
+  def collection = {
+    DB.withConnection { implicit c =>
+      SQL("select * from collection where id = {collection_id}")
+      .on('collection_id -> collectionId).as(Collection.coll.single)
+    }
+  }
+
+  def holds = {
```
Comment (JPrevost) [inline app/models/Item.scala]: maybe `holdCount` would be better? `holds` makes me expect a list of Holds... we also use `xxxCount` in other models (see Publisher) for this type of method.

File: app/controllers/Application.scala (hunk 1 above)
Comment (richardrodgers) [inline app/controllers/Application.scala]: Yes, as discussed, there needs to be much better definition of how the authority for running these functions works - either just SysAdmin, or a 'delegation' model from publisher to SysAdmin

File: app/models/Item.scala (hunk 2 above)
Comment (richardrodgers) [inline app/models/Item.scala]: Absolutely right - the problem is that there are 2 conflicting patterns at work: 'holds' was meant to agree with the Item (method/field) 'transfers' which is a count. So I didn't want 'transfers' & 'holdCount' being the same things and sounding different. All those method names need to be made consistent...

**Comment by JPrevost** [comment]
:+1:

**Pull request merged**